File: LP2/ExpertSystem/employee.py
# =========================================================
# EXPERT SYSTEM
# Employee Performance Evaluation System
# =========================================================

# This expert system evaluates employee
# performance based on:
#
# 1. Attendance
# 2. Project Completion
# 3. Teamwork
#
# The system provides:
# - Excellent Performance
# - Good Performance
# - Average Performance
# - Poor Performance
#
# =========================================================
# HOW TO USE
# =========================================================
#
# 1. Run the program.
#
# 2. Enter employee details.
#
# 3. Provide scores out of 100 for:
#    - Attendance
#    - Project Work
#    - Teamwork
#
# 4. The system calculates average score
#    and displays performance evaluation.
#
# =========================================================


print("===================================")
print(" Employee Performance Expert System ")
print("===================================")


# Take employee name
name = input("\nEnter Employee Name: ")


# =========================================================
# TAKE PERFORMANCE SCORES
# =========================================================

attendance = int(
    input("Enter Attendance Score: ")
)

project = int(
    input("Enter Project Score: ")
)

teamwork = int(
    input("Enter Teamwork Score: ")
)


# =========================================================
# CALCULATE AVERAGE SCORE
# =========================================================

average = (
    attendance +
    project +
    teamwork
) / 3


# =========================================================
# DISPLAY PERFORMANCE RESULT
# =========================================================

print("\n===================================")

print("Employee Name:", name)

print("Average Score:", average)


# =====================================================
# PERFORMANCE EVALUATION
# =====================================================

if average >= 85:

    print("\nPerformance: Excellent")

    print(
        "Suggestion: Eligible for promotion."
    )


elif average >= 70:

    print("\nPerformance: Good")

    print(
        "Suggestion: Consistent performer."
    )


elif average >= 50:

    print("\nPerformance: Average")

    print(
        "Suggestion: Needs improvement."
    )


else:

    print("\nPerformance: Poor")

    print(
        "Suggestion: Requires training."
    )

File: cpp/employee.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {

    cout << "===================================\n";
    cout << " Employee Performance Expert System \n";
    cout << "===================================\n";

    string name;

    int attendance;
    int project;
    int teamwork;

    // Employee Name
    cout << "\nEnter Employee Name: ";
    getline(cin, name);

    // Input Scores
    cout << "Enter Attendance Score: ";
    cin >> attendance;

    cout << "Enter Project Score: ";
    cin >> project;

    cout << "Enter Teamwork Score: ";
    cin >> teamwork;

    // Calculate Average
    double average =
        (attendance + project + teamwork) / 3.0;

    // Display Result
    cout << "\n===================================\n";

    cout << "Employee Name: "
         << name << endl;

    cout << "Average Score: "
         << average << endl;

    // Performance Evaluation
    if (average >= 85) {

        cout << "\nPerformance: Excellent\n";

        cout << "Suggestion: Eligible for promotion.\n";
    }

    else if (average >= 70) {

        cout << "\nPerformance: Good\n";

        cout << "Suggestion: Consistent performer.\n";
    }

    else if (average >= 50) {

        cout << "\nPerformance: Average\n";

        cout << "Suggestion: Needs improvement.\n";
    }

    else {

        cout << "\nPerformance: Poor\n";

        cout << "Suggestion: Requires training.\n";
    }

    return 0;
}
